clipper_admin/test/clipper_manager_test.py

```diff
@@ -0,0 +1,280 @@
+
+    def test_add_container_for_external_model_fails(self):
+        result = self.clipper_inst.add_container(self.model_name,
+                                                 self.model_version_1)
+        self.assertFalse(result)
+
+    def test_model_version_sets_correctly(self):
+        self.clipper_inst.set_model_version(self.model_name,
+                                            self.model_version_1)
+        all_models = self.clipper_inst.get_all_models(verbose=True)
+        models_list_contains_correct_version = False
+        for model_info in all_models:
+            version = model_info["model_version"]
+            if version == self.model_version_1:
+                models_list_contains_correct_version = True
+                self.assertTrue(model_info["is_current_version"])
+
+        self.assertTrue(models_list_contains_correct_version)
+
+    def test_get_logs_creates_log_files(self):
+        log_file_names = self.clipper_inst.get_clipper_logs()
+        self.assertIsNotNone(log_file_names)
+        self.assertGreaterEqual(len(log_file_names), 1)
+        for file_name in log_file_names:
+            self.assertTrue(os.path.isfile(file_name))
+
+    def test_inspect_instance_returns_json_dict(self):
+        metrics = self.clipper_inst.inspect_instance()
+        self.assertEqual(type(metrics), dict)
+        self.assertGreaterEqual(len(metrics), 1)
+
+    def test_model_deploys_successfully(self):
+        # Initialize a support vector classifier 
+        # that will be deployed to a no-op container
+        model_data = svm.SVC()
+        container_name = "clipper/noop-container"
+        labels = ["test"]
+        input_type = "doubles"
+        result = self.clipper_inst.deploy_model(
+            self.deploy_model_name, self.deploy_model_version, model_data,
+            container_name, labels, input_type)
+        self.assertTrue(result)
+        model_info = self.clipper_inst.get_model_info(
+            self.deploy_model_name, self.deploy_model_version)
+        self.assertIsNotNone(model_info)
+        running_containers_output = self.clipper_inst._execute_standard(
+            "docker ps -q --filter \"ancestor=clipper/noop-container\"")
+        self.assertIsNotNone(running_containers_output)
+        self.assertGreaterEqual(len(running_containers_output), 1)
+
+    def test_add_container_for_deployed_model_succeeds(self):
+        result = self.clipper_inst.add_container(self.deploy_model_name,
+                                                 self.deploy_model_version)
+        self.assertTrue(result)
+        running_containers_output = self.clipper_inst._execute_standard(
+            "docker ps -q --filter \"ancestor=clipper/noop-container\"")
+        self.assertIsNotNone(running_containers_output)
+        split_output = running_containers_output.split("\n")
+        self.assertGreaterEqual(len(split_output), 2)
+
+    def test_predict_function_deploys_successfully(self):
```

Comment:
This test will fail if not running inside an anaconda environment. I'm not sure if Jenkins runs inside an anaconda environment or not. I'll look into it but in the meantime if you see the test failing on Jenkins for this reason, that's why it's happening.